### DISCUSSION TITLE
Do we have if/else feature in promql

### DISCUSSION THREAD

**Discussion by ankurnigam** [prometheus/prometheus#9469] [Q&A]
Hi Team ,

I am having one business case where I am getting set of values , I want to develop/record new metric based on certain values of the metrics .

Do we have if/else feature in promql.

Used Case : 

Metric name => my_metric{name="abc"} 
Sample values => 0,1,2,3,4,5,6,7,12,14,16

New metric want to develop

if metric value is 0,1,2,3,4,5
change value to 0
if metric value is 12,14,16
change value to 1

**Comment by roidelapluie** [comment]
This is possible with PromQL already:

`0 * (my_metric{name="abc"} <= 6) or 1+(my_metric{name="abc"} >= 12)`
Based on vector matching, this would work.

Alternative answer: `clamp(my_metric{name="abc"}-10, 0, 1)`. This will reduce the value of my_metric by 10 and make sure the value is not lower than 0 and higher than 1. If the values provided by the example are exact, that would work.
